Modules/File/classes/class.ilObjFile.php

```diff
@@ -678,16 +675,34 @@ function sendFile($a_hist_entry_id = null)
 		if (@is_file($file))
 		{
 			global $ilClientIniFile;
-			
-			// also returning the 'real' filename if a history file is delivered
-			if ($ilClientIniFile->readVariable('file_access','download_with_uploaded_filename') != '1' && is_null($a_hist_entry_id))
-			{
-				ilUtil::deliverFile($file, $this->getTitle(), $this->guessFileType($file), $this->isInline());
+			/**
+			 * @var $ilClientIniFile ilIniFile
+			 */
+			require_once('./Services/FileDelivery/classes/class.ilFileDelivery.php');
+
+			$ilFileDelivery = new ilFileDelivery($file);
+			$ilFileDelivery->setDisposition($this->isInline() ? ilFileDelivery::DISP_INLINE : ilFileDelivery::DISP_ATTACHMENT);
+			$ilFileDelivery->setMimeType($this->guessFileType($file));
+			$ilFileDelivery->setConvertFileNameToAsci(true);
```

Comment:
The disable_ascii setting in the ini file seems to be a setting that could or should affect ILIAS "globally". Imo this option could control the "default" behaviour of ilFileDelivery. Consumers of ilFileDelivery (like ilObjFile) should not need to call setConvertFileNameToAsci() if there is not a very good reason to modify the behaviour. This would "fix" all places that already use ilFileDelivery without the need to call this additional setter.